common/djangoapps/student/admin.py

```diff
@@ -309,9 +312,64 @@ def get_queryset(self, request):
         return super().get_queryset(request).select_related('user')  # lint-amnesty, pylint: disable=no-member, super-with-arguments
 
 
+class LanguageAutocomplete(autocomplete.Select2ListView):
```

Comment:
make sure user is logged in. you can add a decorator here 
`@method_decorator(login_required, name='dispatch')`